Cython/Utility/ObjectHandling.c

```diff
@@ -2311,29 +2239,22 @@ static CYTHON_INLINE PyObject* __Pyx_PyObject_CallNoArg(PyObject *func); /*proto
 
 /////////////// PyObjectCallNoArg ///////////////
 //@requires: PyObjectCallMethO
-//@requires: PyObjectCall
-//@requires: PyFunctionFastCall
-//@substitute: naming
+//@requires: PyObjectFastCall
 
 #if CYTHON_COMPILING_IN_CPYTHON
 static CYTHON_INLINE PyObject* __Pyx_PyObject_CallNoArg(PyObject *func) {
-#if CYTHON_FAST_PYCALL
-    if (PyFunction_Check(func)) {
-        return __Pyx_PyFunction_FastCall(func, NULL, 0);
-    }
-#endif
 #ifdef __Pyx_CyFunction_USED
-    if (likely(PyCFunction_Check(func) || __Pyx_CyFunction_Check(func)))
+    if (PyCFunction_Check(func) || __Pyx_CyFunction_Check(func))
 #else
-    if (likely(PyCFunction_Check(func)))
+    if (PyCFunction_Check(func))
 #endif
     {
         if (likely(PyCFunction_GET_FLAGS(func) & METH_NOARGS)) {
             // fast and simple case that we are optimising for
             return __Pyx_PyObject_CallMethO(func, NULL);
         }
     }
-    return __Pyx_PyObject_Call(func, $empty_tuple, NULL);
+    return __Pyx_PyObject_FastCall(func, NULL, 0);
```

Comment:
I don't think `NULL` is allowed in the non-vectorcall case.

Reply:
I'm pretty sure that it is allowed. For example CPython 3.6 has:
```
#define _PyObject_CallNoArg(func) \
    _PyObject_FastCall((func), NULL, 0)
```